client/src/App.js

```diff
@@ -1,36 +1,57 @@
 import "./App.css";
+import { useState } from "react";
 import { Routes, Route, useLocation } from "react-router-dom";
 import { ChakraProvider, CSSReset } from "@chakra-ui/react";
 
-// COMPONENTS
+// components
 
-import Header, { ADMIN_LINKS, CUSTOMER_LINKS } from "./component/Header";
-import AdminPage from "./component/admin/AdminPage";
-import AdminHome from "./component/admin/AdminHome";
-import MenuListContainer from "./component/admin/MenuListContainer";
-import MessagesPage from "./component/admin/MessagesPage";
-import Menu from "./component/customer/Menu";
-import ContactPage from "./component/customer/ContactPage";
-import ReviewPage from "./component/customer/ReviewPage";
-
-import Footer from "./component/Footer";
-
-import Cart from "./component/customer/Cart";
+import Header, { ADMIN_LINKS, CUSTOMER_LINKS } from "./components/Header";
+import AdminPage from "./components/admin/AdminPage";
+import AdminHome from "./components/admin/AdminHome";
+import MenuListContainer from "./components/admin/MenuListContainer";
+import MessagesPage from "./components/admin/MessagesPage";
+import Home from "./components/customer/Home";
+import ContactPage from "./components/customer/ContactPage";
+import ReviewPage from "./components/customer/ReviewPage";
+import Footer from "./components/Footer";
+import PageNotFound from "./components/PageNotFound";
+import Cart from "./components/customer/Cart";
 
 function App() {
   const location = useLocation();
+  const [cartItems, setCartItems] = useState([]);
   const isAdmin = location.pathname.startsWith("/admin");
+  const addToCart = (item) => {
+    const existingItem = cartItems.find((iterator) => iterator.id === item.id);
+    let updateCartItems;
+    if (existingItem) {
+      existingItem.quantity = item.quantity;
+      updateCartItems = [...cartItems];
```

Comment:
This is mutating state directly. Updating a single item in an array without mutating it is a bit tricky without a helper. Here's one way to do it: https://stackoverflow.com/questions/43792457/update-one-of-the-objects-in-array-in-an-immutable-way